### PR TITLE
Add instructions for using QEMU

### PR DESCRIPTION
**Pull request by swanandx**
It was difficult to find out how to use QEMU for simulation. Therefore I decided to add some steps, so its easier to get it working.

Let me know if any step is ambiguous or in case we need to add more explanation.

### CODE REVIEW

File: src/tooling/simulating/qemu.md [hunk 1]
```diff
@@ -3,6 +3,32 @@
...
 See the [QEMU wiki] for instructions on how to build QEMU and emulate projects with it.
 
+Once you have built QEMU, you should have `qemu-system-xtensa`.
 
 [espressif/QEMU]: https://github.com/espressif/qemu
 [QEMU wiki]: https://github.com/espressif/qemu/wiki
+
+## Running our project using QEMU
+
+> *NOTE*: Only ESP32 is supported at the moment, so make sure you are compiling for `xtensa-esp32-espidf` target.
```
Comment (SergioGasquez) [inline src/tooling/simulating/qemu.md:13]:
```suggestion
> *NOTE*: Only ESP32 is currently supported, so make sure you are compiling for `xtensa-esp32-espidf` target.
```

File: src/tooling/simulating/qemu.md [hunk 2]
```diff
@@ -3,6 +3,32 @@
...
+
+For running our project in QEMU, we need a firmware/image which has bootloader and partition table merged in it.
```
Comment (SergioGasquez) [inline src/tooling/simulating/qemu.md:15]:
```suggestion
For running our project in QEMU, we need a firmware/image with bootloader and partition table merged. 
```

File: src/tooling/simulating/qemu.md [hunk 3]
```diff
@@ -3,6 +3,32 @@
...
 
 [espressif/QEMU]: https://github.com/espressif/qemu
 [QEMU wiki]: https://github.com/espressif/qemu/wiki
+
+## Running our project using QEMU
+
+> *NOTE*: Only ESP32 is supported at the moment, so make sure you are compiling for `xtensa-esp32-espidf` target.
+
+For running our project in QEMU, we need a firmware/image which has bootloader and partition table merged in it. 
+We can use [cargo-espflash] to generate it:
```
Comment (SergioGasquez) [inline src/tooling/simulating/qemu.md:16]:
```suggestion
We can use [`cargo-espflash`] to generate it:
```

File: src/tooling/simulating/qemu.md [hunk 4]
```diff
@@ -3,6 +3,32 @@
...
+
+[cargo-espflash]: https://github.com/esp-rs/espflash
```
Comment (SergioGasquez) [inline src/tooling/simulating/qemu.md:18]:
```suggestion
[`cargo-espflash`]: https://github.com/esp-rs/espflash/tree/main/cargo-espflash
```

File: src/tooling/simulating/qemu.md [hunk 5]
```diff
@@ -3,6 +3,32 @@
...
+```bash
+cargo espflash save-image --merge ESP32 <OUTFILE> --release
+```
+
+> If you prefer to use `espflash`, you can achieve same by building project first and then generating image:
+> ```bash
+> cargo build --release
+> espflash save-image --merge ESP32 <OUTFILE> target/xtensa-esp32-espidf/release/<NAME>
+> ```
+
```
Comment (SergioGasquez) [inline src/tooling/simulating/qemu.md:29]:
```suggestion

[`espflash`]: https://github.com/esp-rs/espflash/tree/main/espflash
```

File: src/tooling/simulating/qemu.md [hunk 6]
```diff
@@ -3,6 +3,32 @@
...
+For running our project in QEMU, we need a firmware/image which has bootloader and partition table merged in it. 
+We can use [cargo-espflash] to generate it:
+
+[cargo-espflash]: https://github.com/esp-rs/espflash
+
+```bash
+cargo espflash save-image --merge ESP32 <OUTFILE> --release
+```
+
+> If you prefer to use `espflash`, you can achieve same by building project first and then generating image:
```
Comment (SergioGasquez) [inline src/tooling/simulating/qemu.md:24]:
```suggestion
> If you prefer to use [`espflash`], you can achieve the same result by building the project first and then generating image:
```

File: src/tooling/simulating/qemu.md [hunk 7]
```diff
@@ -3,6 +3,32 @@
...
+> ```bash
+> cargo build --release
+> espflash save-image --merge ESP32 <OUTFILE> target/xtensa-esp32-espidf/release/<NAME>
+> ```
+
+Now, run the image in QEMU :
```
Comment (SergioGasquez) [inline src/tooling/simulating/qemu.md:30]:
```suggestion
Now, run the image in QEMU:
```

**Review by SergioGasquez — COMMENTED** [review]
Hi! Thanks for your contribution! This was a pending task (#40), but since I had no experience with QEMU, it was empty, until now! I left a few (mostly irrelevant) comments, feel free to address them.

@KaryFeever do you mind having a look at this PR (if you have some free time)?

Also, we are planning a major upgrade to the book (see #59). Of course, this does not mean that we will discard this content, but it might get modified or moved!

Thanks again!

**Comment by swanandx** [comment]
Hey, thanks for review! I also don't have ( didn't had ) experience with QEMU, I wrote this while I am still learning, so it would be nice to get it reviewed by someone who does have experience with QEMU haha.
 
 > Also, we are planning a major upgrade to the book (see https://github.com/esp-rs/book/issues/59). Of course, this does not mean that we will discard this content, but it might get modified or moved!
 
That is good to know! Hope the new version of book comes soon <3 Let me know if there is something I can help with, as I just started my journey with embedded rust and esp, I would love to contribute as I learn.

**Review by igrr — APPROVED** [review]

**Pull request merged**